src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,39 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationFailedException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int DEFAULT_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new ValidationFailedException("User is null");
+        }
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationFailedException("This login is already taken: " + user.getLogin());
+        }
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new ValidationFailedException("Login and password can't be null");
+        }
+        if (user.getLogin().length() < DEFAULT_LENGTH
+                || user.getPassword().length() < DEFAULT_LENGTH) {
+            throw new ValidationFailedException("Login and password must be at least 6 characters");
+        }
+        if (user.getAge() == null) {
+            throw new ValidationFailedException("Age can't be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new ValidationFailedException("Can't register your age is below 18");
```

Comment:
The exception message should include the actual age of the user to be informative, as mentioned in the good example of the checklist.